### PR TITLE
QP-Stargate-Fee

### PR DESCRIPTION
**Pull request by raviansalman**

### CODE REVIEW

File: .env.example
Comment (lukmaan-ferrum) [inline .env.example:1]: can we add this file back in lol

File: contracts/FiberRouter.sol [hunk 1]
```diff
@@ -24,6 +25,7 @@ contract FiberRouter is FeeDistributor, QuantumPortalApp, CCIPApp {
         address sourceFoundryToken,
         uint256 amountIn,
         uint256 feeAmount,
+        address sender,
```
Comment (lukmaan-ferrum) [inline contracts/FiberRouter.sol]: does it work if we get rid of this input and just use msg.sender?

File: contracts/FiberRouter.sol [hunk 2]
```diff
@@ -146,6 +152,8 @@ contract FiberRouter is FeeDistributor, QuantumPortalApp, CCIPApp {
             _bridgeWithPortal(dstChainId, recipient, sourceFoundryToken, amountIn, feeAmount, dstData);
         } else if (swapType == 1) {
             _bridgeWithCcip(dstChainId, sourceFoundryToken, amountIn, abi.encode(recipient));
+        } else if (swapType == 2) {
+            _bridgeWithStargate(amountIn, sender, recipient, dstChainId);
```
Comment (lukmaan-ferrum) [inline contracts/FiberRouter.sol]: so msg.sender instead of sender input?

File: hardhat.config.ts
Comment (lukmaan-ferrum) [inline hardhat.config.ts:1]: I think just copy paste my hardhat.config.ts. So far we have been exposing our paid rpc endpoints 😅

**Review by lukmaan-ferrum — CHANGES_REQUESTED** [review]

File: contracts/FiberRouter.sol (hunk 1 above)
Comment (raviansalman) [reply contracts/FiberRouter.sol]: If we are directly using fiberRouter as the first contract to interact with then msg.sender will be the sourceWallet. So, yes we can use msg.sender in that case.

File: hardhat.config.ts
Comment (raviansalman) [reply hardhat.config.ts:1]: sure sure

**Review by lukmaan-ferrum — APPROVED** [review]

**Pull request merged**